pkg/services/publicdashboards/service/query.go

```diff
@@ -37,8 +34,8 @@ func (pd *PublicDashboardServiceImpl) FindAnnotations(ctx context.Context, reqDT
 		return nil, models.ErrInternalServerError.Errorf("FindAnnotations: failed to unmarshal dashboard annotations: %w", err)
 	}
 
-	anonymousUser := buildAnonymousUser(ctx, dash, pd.features)
-
+	// We don't have a signed in user for public dashboards. We are using Grafana's Identity to query the annotations.
```

Comment:
It's because we end up making a query for the dashboard:

https://github.com/grafana/grafana/blob/a5ce4a6aad3f9a5a040abf2968b255d17796ad16/pkg/services/annotations/accesscontrol/accesscontrol.go#L73
https://github.com/grafana/grafana/blob/a5ce4a6aad3f9a5a040abf2968b255d17796ad16/pkg/services/annotations/accesscontrol/accesscontrol.go#L139
https://github.com/grafana/grafana/blob/a5ce4a6aad3f9a5a040abf2968b255d17796ad16/pkg/services/dashboards/service/dashboard_service.go#L1247